### ISSUE TITLE
[Accessibility][Headings and Labels - Property Pane_Ask a question]: Heading structure is not correct as the name of the pane should be defined under heading tags.

### ISSUE THREAD

**Issue by ashish315** [microsoft/BotFramework-Composer#2080]
**User Experience** 
Users who rely on AT will face difficulty and easily confused if the headings are not defined correctly.

**Test Environment**
Browser: Microsoft Edge Canary Version: 82.0.425.0 (Official build)
OS build: 2004 (19564.1005)
Screen Reader: Narrator
URL: [Bot Framework Composer](http://localhost:3000/dialogs/Main)
[**Prerequisite Here**](http://localhost:3000/dialogs/Main?selected=triggers[5]&focused=triggers[5].actions[0])

**Repro Steps:** 
1. Open the above mentioned URL in EDGE and turn on narrator.
2. Navigate to already created dialog tree item and activate it. Such as- Greetings, Actions, EndDialog etc.
3. Click on "+" and select 'Ask a question'. Then select any of the available options except Oauth login.
4. Navigate to property pane appeared at right side of the screen and navigate with 'H' i.e. shortcut for Headings.
5. Observer the issue in Property Pane.

**Actual:** 
The heading of the Property Pane i.e. 'Prompt for text' edit field is not defined under heading tags.

**Expected:**
The name of the Property Pane should be defined under heading tag. So, that the user will able to navigate quickly through the heading shortcut. 'Prompt for text' edit field should be defined under heading level 1 or level i.e as <h1> or <h2>.

**Note:**
Issue is observed for all the Property Pane  appeared at right side of the screen

**MAS Impacted:** [MAS2.4.6](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={0cf80872-800d-4643-90cf-0c1c3d3e6260})

**Attachment:**
![MAS2 4 6_PropertyPane](https://user-images.githubusercontent.com/60913131/75242581-69f94c00-57ee-11ea-857d-6ea62137a1eb.png)

**Comment by corinagum** [comment]
@DesignPolice  Looping you in; Inputs cannot have headings in them, so it's possible the input location would need to be changed in favor of the heading. Does this require design input? (I'm assuming yes).

If I'm wrong, feel free to unassign yourself :P

**Comment by DesignPolice** [comment]
Good morning. Happy to help, I'm trying to duplicate the issue, to make a decision on it... In the screen grab, Prompt for text with Prompt for text under it is odd, but is this saying there should be a header above it called Prompt of Text instead of under? 

I'm seeing this.. OAuth login - but nothing with prompt. So I assume I'm doing it wrong.

<img width="408" alt="Screen Shot 2020-02-27 at 8 22 00 AM" src="https://user-images.githubusercontent.com/45223620/75463158-44795900-593a-11ea-8f55-9a4e9aeb69fe.png">

Anyway.

They are saying we need a header above Prompt for text - which makes sense, but I wouldn't want a header to say Prompt for Text, then a Fill in the blank Prompt for Text and then below it again... 

@corinagum

**Comment by corinagum** [comment]
Ah, directions say pick any option except OAuth option. Could you take a look at a different one? Let me know if you want to call over Teams

**Comment by corinagum** [comment]
@DesignPolice ping on this guy

**Comment by DesignPolice** [comment]
alright ya that is on me - will take to the 11am meeting today - thanks

**Comment by mareekuh** [comment]
From what I am seeing in the initial screenshot, it looks like there is a delta between the sections as we see them and have labeled them from what the Screen Reader sees/reads.
So, specifically with respect to the Properties pane: The pane should be labeled 'Properties (pane)' and the title shown provides the specifics: Text Input Properties.

@cwhitten - from a dev side, how can we make sure the UX-area grouping and labels are reflected correctly in accessibilty mode/through the screen reader? 

 more info...

Also:
This document captures how the areas in the UX are grouped and labeled:

Note: for the 'design flow', the 'filter box' should be part of the toolbar vs part of the nav. the nav is where the results show.

https://github.com/MicrosoftDocs/composer-docs-pr/blob/SM-NamingConventions-Composer/contributor-guide/Naming%20conventions%20for%20the%20Bot%20Framework%20Composer.md

![Screen Shot 2020-04-10 at 4 39 00 PM](https://user-images.githubusercontent.com/48503434/79029583-d388ac00-7b49-11ea-8ede-ab8d7343f876.png)

![Screen Shot 2020-04-10 at 4 38 37 PM](https://user-images.githubusercontent.com/48503434/79029585-d4b9d900-7b49-11ea-8425-5bf0ec65568d.png)
![Screen Shot 2020-04-10 at 4 38 27 PM](https://user-images.githubusercontent.com/48503434/79029587-d4b9d900-7b49-11ea-9b06-296ce4dfe03b.png)

**Comment by beyackle** [comment]
@cwhitten @zxyanliu @DesignPolice Where are we on this? If there's something that needs fixing I'd like to be able to clear this ticket out, but I need guidance on what.

**Comment by DesignPolice** [comment]
Hey Ben, 

Yes this is ready to go, from what I was able to understand, is that in the code on the site these areas need to have titles for a Screen Reader. @mareekuh outlined the areas and gave them names, The visual key is just to make sure you can tell what area is what.

I will be around if you have more questions, just ping me, I'm working on Build stuff.

**Comment by beyackle** [comment]
Okay, I see exactly what needs doing now. I'll knock this out in the morning.

**Comment by beyackle** [comment]
This is how it ends up looking in Narrator's list of landmarks: 
![image](https://user-images.githubusercontent.com/61990921/81222418-17c86a00-8f99-11ea-9142-92878513033f.png)

**Comment by mareekuh** [comment]
What is 'navigation' the left-most panel? If so, we call that the Menu.
Main = the title/header of the page (Settings, Bot Responses etc)? If so, let's call that 'Title'
What is the 'navigation path'?

Please use **S**entence **c**apitalization

**Comment by DesignPolice** [comment]
I have been having too many access issues today... So sadly here are the screenshots from the Teams chat that Ben and I had.

We just need to define areas and labels for the other screens as well. 

<img width="697" alt="Screen Shot 2020-05-06 at 5 14 29 PM" src="https://user-images.githubusercontent.com/45223620/81240892-96cf9980-8fbd-11ea-86d5-185e4eda49e4.png">

<img width="714" alt="Screen Shot 2020-05-06 at 5 14 44 PM" src="https://user-images.githubusercontent.com/45223620/81240898-9b944d80-8fbd-11ea-8925-890f08a0373e.png">

<img width="792" alt="Screen Shot 2020-05-06 at 5 14 52 PM" src="https://user-images.githubusercontent.com/45223620/81240902-9f27d480-8fbd-11ea-824d-8648565d39e6.png">

**Comment by mareekuh** [comment]
**User input**
- left nav = Navigation panel
- Title
- body of the page = List view

**Notifications**
(I asked Chris to turn the dropdown into a leftnav to be parallel to the other pages
- Title
- List view (The columns can be referred to the column names)

**Publish profiles** (note: sentence cap!)
- Navigation panel
- Title
- List view

**Skills**
hmm. Listview and settings?
Next round this page will look like the others: Navigation panel and Listview
- Navigation panel
- Title
- List view


**Settings**
- Navigation panel
- Title
- List view

**Comment by zxyanliu** [comment]
FYI. This is [the naming convention](https://github.com/MicrosoftDocs/composer-docs-pr/blob/SM-NamingConventions-Composer/contributor-guide/Naming%20conventions%20for%20the%20Bot%20Framework%20Composer.md) shared in our writers' team. The screenshots are not up to date, but the naming and labeling of the areas might be helpful. :)

* Bot Responses
![bot-responses](https://user-images.githubusercontent.com/32497439/81242452-ffb91080-8fc1-11ea-8464-fda99e30bada.png)

* User Input
![user-input](https://user-images.githubusercontent.com/32497439/81242420-e7e18c80-8fc1-11ea-8cfa-2fb5c419a963.png)

* Notifications
![notifications](https://user-images.githubusercontent.com/32497439/81242470-0f385980-8fc2-11ea-85d9-cdd8193d4ed7.png)

* Settings
![settings](https://user-images.githubusercontent.com/32497439/81242474-1495a400-8fc2-11ea-84e7-93324b74e024.png)

**Comment by mareekuh** [comment]
A few comments:
**User input** 

@cwhitten @zxyanliu @DesignPolice @beyackle 

User input - Do we need to mark the 'title'?
- What do we call the view if the editor switch is turned on vs turned off? If it's on, 'User input Editor' makes sense to me, when on, User Input listview makes sense

**Notifications** - please use 'page control' as a label vs navigation control. We'll be moving the 'all-warnings- errors' dropdown into the navigation pane and the page control will remain at the bottom or go somewhere else.
Settings - this page will be undergoing some UI changes also:

- 'Settings' will in the 'title' area, just like the publishing page etc.
- the settings menu will be looking the same as all of the left navs, so we can call this 'settings navigation pane' and it should take the same visual conventions as all other leftnavs in the app (right now 'configuration' has a special treatment which we'll want to remove'
- Settings editor works for me if the editor switch is on. We're also planning to have a GUI version of the settings page

**Comment by beyackle** [comment]
Re-opened due to #2956 . Will redo.

**Comment by mareekuh** [comment]
Please let’s keep things consistent and call it the :

  *   Bot Responses Navigation pane and editor?
  *   User Input Navigation pane and editor?

Also

  *   Notifications:  Paging control (vs navigation control as moving forward we are turning the ‘all/errors/notifications dropdown into a left nav panel, like the other pages?
  *   Settings menu = Settings navigation panel

**Comment by mareekuh** [comment]
@zxyanliu

**Comment by zxyanliu** [comment]
@mareekuh Yes. We should make the terms consistent. I will update the naming conventions as suggested. Thanks!

**Comment by beyackle** [comment]
I've updated the labels in #2959 as well.

**Comment by ashish315** [comment]
@cwhitten As checked the only one heading i.e. chart name is defined as heading still the property pane didn't have any headings.

**Comment by zxyanliu** [comment]
Are we still working on this ticket? What is the status? What is needed from the docs?

**Comment by beyackle** [comment]
We implemented navigation landmarks (https://accessibility.oit.ncsu.edu/it-accessibility-at-nc-state/developers/accessibility-handbook/aria-landmarks/) as a way for SR users to navigate around the page; this is IMO a better option than messing up the visual design by putting literal headings everywhere. I'd be okay closing this.

**Comment by zxyanliu** [comment]
Hey @beyackle, thanks for the info. 
If there is nothing required on the docs side, I am going to unassign myself from this thread.

**Comment by Amit8527510735** [comment]
#Nonbenchmark;
